services/src/kilda-core/kilda-persistence-neo4j/src/main/java/org/openkilda/persistence/repositories/impl/Neo4jIslRepository.java

```diff
@@ -0,0 +1,98 @@
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ */
+
+package org.openkilda.persistence.repositories.impl;
+
+import org.openkilda.model.Isl;
+import org.openkilda.model.SwitchId;
+import org.openkilda.persistence.repositories.IslRepository;
+
+import org.neo4j.ogm.cypher.query.SortOrder;
+
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Neo4J OGM implementation of {@link IslRepository}.
+ */
+public class Neo4jIslRepository extends Neo4jGenericRepository<Isl> implements IslRepository {
+    public Neo4jIslRepository(Neo4jSessionFactory sessionFactory) {
+        super(sessionFactory);
+    }
+
+    @Override
+    public Isl findByEndpoint(SwitchId switchId, int port) {
+        Map<String, Object> parameters = new HashMap<>();
+        parameters.put("src_switch", switchId.toString());
+        parameters.put("src_port", port);
+
+        return getSession().queryForObject(Isl.class, "MATCH (src:switch)-[target:isl]->(dst:switch)\n"
+                + " WHERE src.name=$src_switch AND target.src_port=$src_port\n"
+                + " RETURN src, target, dst", parameters);
+    }
+
+    @Override
+    public Iterable<Isl> findActiveAndOccupiedByFlow(String flowId) {
+        // 0 bandwidth means ignore it.
+        return findActiveAndOccupiedByFlowWithAvailableBandwidth(flowId, 0);
+    }
+
+    @Override
+    public Iterable<Isl> findActiveAndOccupiedByFlowWithAvailableBandwidth(String flowId, long requiredBandwidth) {
+        Map<String, Object> parameters = new HashMap<>();
+        parameters.put("flow_id", flowId);
+        parameters.put("requested_bandwidth", requiredBandwidth);
+
+        String query = "MATCH (src:switch)-[fs:flow_segment{flowid: $flow_id}]->(dst:switch) "
+                + "MATCH (src)-[link:isl]->(dst) "
+                + "WHERE src.state = 'active' AND dst.state = 'active' AND link.status = 'active' "
+                + " AND link.src_port = fs.src_port AND link.dst_port = fs.dst_port "
+                + " AND (link.available_bandwidth + fs.bandwidth >= $requested_bandwidth) "
+                + "RETURN src, link, dst";
+
+        return getSession().query(Isl.class, query, parameters);
+    }
+
+    @Override
+    public Iterable<Isl> findActive() {
+        // 0 bandwidth means ignore it.
+        return findActiveWithAvailableBandwidth(0L);
+    }
+
+    @Override
+    public Iterable<Isl> findActiveWithAvailableBandwidth(long requiredBandwidth) {
+        Map<String, Object> parameters = new HashMap<>();
+        parameters.put("requested_bandwidth", requiredBandwidth);
+
+        String query = "MATCH (src:switch)-[link:isl]->(dst:switch) "
+                + "WHERE src.state = 'active' AND dst.state = 'active' AND link.status = 'active' "
+                + " AND src.name IS NOT NULL AND dst.name IS NOT NULL "
```

Comment:
Why do we check switch "names" here? Is the ISL can be attached to the switch with `.state == "active"` and without name?

Reply:
Good catch. The legacy query. Fixed - check for null has been removed.